glsk/glsk-document-cse/src/main/java/com/powsybl/glsk/cse/CseGlskDocument.java

```diff
@@ -9,70 +9,155 @@
+
+            // Setup schema validator
+            SchemaFactory sf = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+            URL glskSchemaResource = CseGlskDocument.class.getResource("/xsd/gsk-document.xsd");
+            if (glskSchemaResource != null) {
+                Schema glskSchema = sf.newSchema(new File(glskSchemaResource.getFile()));
+                unmarshaller.setSchema(glskSchema);
+            } else {
+                LOGGER.warn("Unable to find GLSK Schema definition file. GLSK file will be imported without schema validation.");
+            }
+
+            // Unmarshal xml file
+            GSKDocument nativeGskDocument = (GSKDocument) JAXBIntrospector.getValue(unmarshaller.unmarshal(inputStream));
+            return new CseGlskDocument(nativeGskDocument);
+        } catch (SAXException e) {
+            throw new GlskException("Unable to import CSE GLSK file: Schema validation failed.", e);
+        } catch (JAXBException e) {
             throw new GlskException("Unable to import CSE GLSK file.", e);
         }
     }
 
-    private CseGlskDocument(Document document) {
-        NodeList timeSeriesNodeList = document.getElementsByTagName("TimeSeries");
-        for (int i = 0; i < timeSeriesNodeList.getLength(); i++) {
-            if (timeSeriesNodeList.item(i).getNodeType() == Node.ELEMENT_NODE) {
-                Element timeSeriesElement = (Element) timeSeriesNodeList.item(i);
-                GlskPoint glskPoint = new CseGlskPoint(timeSeriesElement);
+    private CseGlskDocument(GSKDocument nativeGskDocument) {
+        // Computation of "standard" and "export" timeseries
+        Map<String, List<GlskPoint>> cseGlskPointsStandard = getGlskPointsFromTimeSeries(nativeGskDocument.getTimeSeries());
+        Map<String, List<GlskPoint>> cseGlskPointsExport = getGlskPointsFromTimeSeries(nativeGskDocument.getTimeSeriesExport());
+
+        if (calculationDirectionsAbsent(nativeGskDocument)) {
+            // "default" mode : all countries are considered in full import (use TimeSeries for all)
+            this.cseGlskPoints.putAll(cseGlskPointsStandard);
+        } else {
+            // Extract CalculationDirections
+            List<CalculationDirectionType> calculationDirections = nativeGskDocument.getCalculationDirections().get(0).getCalculationDirection();
+            Map<String, List<String>> countriesInAndOutArea = getCountriesInAndOutArea(calculationDirections);
+
+            // Use data from cseGlskPointsStandard or cseGlskPointsExport depending on CalculationDirections
+            fillGlskPointsForExportCorner(cseGlskPointsStandard, cseGlskPointsExport, countriesInAndOutArea);
+        }
+    }
+
+    private static Map<String, List<GlskPoint>> getGlskPointsFromTimeSeries(List<TimeSeriesType> timeSeries) {
+        Map<String, List<GlskPoint>> cseGlskPoints = new TreeMap<>();
+        if (timeSeries == null) {
+            return cseGlskPoints;
```

Comment:
An exception, I'm sure not.
A warning, not sure if it is useful or not... Theoretically the `timeSeries` variable here should never be null, as we check  in `CseGlskDocumentImporter.canImport()` method  that the document contains at least one element named <TimeSeries>.  It could possibly be null for timeSeriesExport, if no <TimeSeriesExport> markup exists in the document. But it will happen for every "standard" GLSK file handling (I mean here, no Export Corner handling), so I don't think we need to warn the user systematically.